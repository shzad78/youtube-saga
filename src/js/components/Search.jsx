import React from 'react';
import PropTypes from 'prop-types';

export default class Search extends React.Component {
  constructor() {
    super();

    this.state = {
      value: ''
    };

    this.localHandler = this.localHandler.bind(this);
  }

  localHandler(e) {
    const { getVideos } = this.props;

    this.setState({ value: e.target.value });
    getVideos(e.target.value);
  }

  render() {
    return (
      <div className="search col-md-6 col-md-offset-4 form-inline">
        <input
          type="text"
          className="form-control"
          onChange={this.localHandler}
          value={this.state.value}
        />
        <button className="btn">
          <span className="glyphicon glyphicon-search " />
        </button>
      </div>
    );
  }
}

Search.propTypes = {
  getVideos: PropTypes.func.isRequired
};
