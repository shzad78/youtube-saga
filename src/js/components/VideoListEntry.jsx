import React from 'react';
import PropTypes from 'prop-types';

const VideoListEntry = ({video, handleSelectedVideo}) => {

  return (
    <div className="videoListEntry"> 
      <div className="media">
        <div className="media-left media-middle">
          <img 
            className="media-object" 
            src={video.snippet.thumbnails.default.url} 
          />
        </div>
        <div className="media-body">
          <h4 
            className="media-heading"
            onClick={() => handleSelectedVideo(video)}
          >
            {video.snippet.title}
          </h4>
          <p className="media-description">{video.snippet.description}</p>
        </div>
      </div>
    </div>
  )
};

VideoListEntry.propTypes = {
  video: PropTypes.object.isRequired,
  handleSelectedVideo: PropTypes.func.isRequired,
}

export default VideoListEntry;

