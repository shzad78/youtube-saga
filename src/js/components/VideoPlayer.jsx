import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

export const VideoPlayer = ({selectedVideo}) => {
  if(Object.keys(selectedVideo).length) {
    var selectedVideoURL = "https://www.youtube.com/embed/" + selectedVideo.id.videoId
  }

  return (
    <div className="videoPlayer"> 
      <div className="player embed-responsive embed-responsive-16by9">
        <iframe 
          className="embed-responsive-item" 
          src={selectedVideoURL}
          allowFullScreen
        />
      </div>
      <div className="video-details">
        <h4 className="video-title">
        {Object.keys(selectedVideo).length ? selectedVideo.snippet.title : ""}
        </h4>
        <p className="video-description">
        {Object.keys(selectedVideo).length ? selectedVideo.snippet.description : ""}
        </p>
      </div>
    </div>
  )
};

VideoPlayer.propTypes = {
  selectedVideo: PropTypes.object.isRequired,
}

function mapStateToProps({selectedVideo}) {
  return {
    selectedVideo
  };
};

export default connect(mapStateToProps)(VideoPlayer);



