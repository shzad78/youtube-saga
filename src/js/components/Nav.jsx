import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Search from './Search';
import { getVideos } from '../actions';

export const Nav = props => (
  <div className="nav">
    <Search {...props} />
  </div>
);

Nav.propTypes = {
  getVideos: PropTypes.func.isRequired
};

export default connect(null, { getVideos })(Nav);
