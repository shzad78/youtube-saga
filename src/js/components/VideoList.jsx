import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import VideoListEntry from './VideoListEntry';
import { selectVideo } from '../actions';

export class VideoList extends React.Component {

  createList() {
    let { videoList, selectVideo } = this.props;

    return videoList.map((item, i) => {
      return <VideoListEntry key={i} video={item} handleSelectedVideo={selectVideo}/>
    });
  }

  render() {
    let { videoList, isLoading, error } = this.props;

    return (
      <div className="videoList"> 
         { 
            isLoading ? <div>Loading Videos.....</div> :
            error ? <div>{error.message}</div> : this.createList()
         }
      </div>
    )
  }
};

VideoList.propTypes = {
  videoList: PropTypes.array.isRequired,
  selectVideo: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  error: PropTypes.object
}

function mapStateToProps({videoList}) {
  return {
    videoList: videoList.videos,
    isLoading: videoList.isLoading,
    error: videoList.error
  }
};

export default connect(mapStateToProps, {selectVideo})(VideoList);
