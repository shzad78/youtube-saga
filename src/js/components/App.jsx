import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios'; 

import Nav from './Nav';
import VideoPlayer from './VideoPlayer';
import VideoList from './VideoList';
import { YOUTUBE_KEY } from '../config/youtubekey';
import { getVideos } from '../actions';


export class App extends React.Component {

  render() {

    return (
      <div className="app">
        <Nav />
        <div className="col-md-7">
          <VideoPlayer />
        </div>
        <div className="col-md-5">
          <VideoList />
        </div>
      </div>
    )
  }

  componentDidMount() {
    const { getVideos } = this.props;

    getVideos('javascript', true);
  }
};

export default connect(null, {getVideos})(App);







