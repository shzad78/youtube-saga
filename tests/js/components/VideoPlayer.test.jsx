import React from 'react';
import { VideoPlayer } from '../../../src/js/components/VideoPlayer';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });


describe('Testing VideoPlayer Component',()=>{
  
  it('renders the VideoPlayer Component', ()=>{ 
    let video = {
      id: {
        videoId: 1234
      },
      snippet: {
        title: 'some title',
        description: 'some description'
      }
    }
    let wrapper = shallow(<VideoPlayer selectedVideo={video} />); 
    expect(wrapper).toMatchSnapshot();
  });
  
  it('it renders given an empty video object', ()=>{
    let video = {};
    let wrapper = shallow(<VideoPlayer selectedVideo={video} />); 
    expect(wrapper).toMatchSnapshot();
  });

});


