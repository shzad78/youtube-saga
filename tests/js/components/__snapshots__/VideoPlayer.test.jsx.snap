// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoPlayer Component it renders given an empty video object 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    selectedVideo={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="player embed-responsive embed-responsive-16by9"
        >
          <iframe
            allowFullScreen={true}
            className="embed-responsive-item"
            src={undefined}
          />
        </div>,
        <div
          className="video-details"
        >
          <h4
            className="video-title"
          >
            
          </h4>
          <p
            className="video-description"
          >
            
          </p>
        </div>,
      ],
      "className": "videoPlayer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            allowFullScreen={true}
            className="embed-responsive-item"
            src={undefined}
          />,
          "className": "player embed-responsive embed-responsive-16by9",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "allowFullScreen": true,
            "className": "embed-responsive-item",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="video-title"
            >
              
            </h4>,
            <p
              className="video-description"
            >
              
            </p>,
          ],
          "className": "video-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "video-title",
            },
            "ref": null,
            "rendered": "",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "video-description",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="player embed-responsive embed-responsive-16by9"
          >
            <iframe
              allowFullScreen={true}
              className="embed-responsive-item"
              src={undefined}
            />
          </div>,
          <div
            className="video-details"
          >
            <h4
              className="video-title"
            >
              
            </h4>
            <p
              className="video-description"
            >
              
            </p>
          </div>,
        ],
        "className": "videoPlayer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              allowFullScreen={true}
              className="embed-responsive-item"
              src={undefined}
            />,
            "className": "player embed-responsive embed-responsive-16by9",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "allowFullScreen": true,
              "className": "embed-responsive-item",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="video-title"
              >
                
              </h4>,
              <p
                className="video-description"
              >
                
              </p>,
            ],
            "className": "video-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "video-title",
              },
              "ref": null,
              "rendered": "",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "video-description",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Testing VideoPlayer Component renders the VideoPlayer Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoPlayer
    selectedVideo={
      Object {
        "id": Object {
          "videoId": 1234,
        },
        "snippet": Object {
          "description": "some description",
          "title": "some title",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="player embed-responsive embed-responsive-16by9"
        >
          <iframe
            allowFullScreen={true}
            className="embed-responsive-item"
            src="https://www.youtube.com/embed/1234"
          />
        </div>,
        <div
          className="video-details"
        >
          <h4
            className="video-title"
          >
            some title
          </h4>
          <p
            className="video-description"
          >
            some description
          </p>
        </div>,
      ],
      "className": "videoPlayer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            allowFullScreen={true}
            className="embed-responsive-item"
            src="https://www.youtube.com/embed/1234"
          />,
          "className": "player embed-responsive embed-responsive-16by9",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "allowFullScreen": true,
            "className": "embed-responsive-item",
            "src": "https://www.youtube.com/embed/1234",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="video-title"
            >
              some title
            </h4>,
            <p
              className="video-description"
            >
              some description
            </p>,
          ],
          "className": "video-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some title",
              "className": "video-title",
            },
            "ref": null,
            "rendered": "some title",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "some description",
              "className": "video-description",
            },
            "ref": null,
            "rendered": "some description",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="player embed-responsive embed-responsive-16by9"
          >
            <iframe
              allowFullScreen={true}
              className="embed-responsive-item"
              src="https://www.youtube.com/embed/1234"
            />
          </div>,
          <div
            className="video-details"
          >
            <h4
              className="video-title"
            >
              some title
            </h4>
            <p
              className="video-description"
            >
              some description
            </p>
          </div>,
        ],
        "className": "videoPlayer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              allowFullScreen={true}
              className="embed-responsive-item"
              src="https://www.youtube.com/embed/1234"
            />,
            "className": "player embed-responsive embed-responsive-16by9",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "allowFullScreen": true,
              "className": "embed-responsive-item",
              "src": "https://www.youtube.com/embed/1234",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="video-title"
              >
                some title
              </h4>,
              <p
                className="video-description"
              >
                some description
              </p>,
            ],
            "className": "video-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "some title",
                "className": "video-title",
              },
              "ref": null,
              "rendered": "some title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "some description",
                "className": "video-description",
              },
              "ref": null,
              "rendered": "some description",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
